source/manual/how-to-remove-jquery-from-javascript.html.md

```diff
@@ -0,0 +1,202 @@
+
+```javascript
+// jQuery (old)
+var module = new GOVUK.Modules.nameOfModule.start(jQueryElement)
+module.aFunction()
+
+// JS (new)
+var module = new GOVUK.Modules.nameOfModule(jsElement)
+module.init()
+module.aFunction()
+```
+
+If the module code is not called again, this can be written more simply as:
+
+```javascript
+new GOVUK.Modules.nameOfModule.start(jQueryElement) // jQuery (old)
+new GOVUK.Modules.nameOfModule(jsElement).init() // JS (new)
+```
+
+## Resources
+
+- [youmightnotneedjquery.com](http://youmightnotneedjquery.com/)
+- the `#govuk-frontenders` or `#frontend` slack channels
+
+## Common problems
+
+### jQuery objects are different from JS nodes
+
+Using jQuery to find an element returns a jQuery object that can have jQuery functions applied to it e.g.
+
+```javascript
+var jqExample = $('div')
+jqExample.addClass('test')
+```
+
+Using JS to find an element returns a node, which can have JS functions applied to it e.g.
+
+```javascript
+var jsExample = document.querySelector('div')
+jsExample.addClass('test') // errors
+jsExample.classList.add('test') // works
+```
+
+`jqExample` and `jsExample` are incompatible. jQuery functions cannot be applied to the JS node and vice versa.
+
+The jQuery object contains a JS node, accessed as `jqExample[0]`. Note that `jqExample` is not an iterable array.
+
+### Things where jQuery and JS use the same function name
+
+Some jQuery functions are named the same as their equivalent JS function, for example:
+
+- `click()`
+- `focus()`
+- `closest()`
+
+This is confusing, because the code chooses the right one to use based on what kind of object it’s being run on.
+
+```javascript
+$('button').click() // calls the jQuery function
+document.querySelector('button').click() // calls the JS
+```
+
+Note that some of these may behave slightly differently, particularly in what they return.
+
+### jQuery is more forgiving than JS
+
+The result of jQuery’s find function might not find anything, but it will still return an empty jQuery object. Subsequent actions on that object (e.g. adding a class) might not do anything but they won’t cause an error.
+
+Attempting the same thing with JS and `querySelector` will return `null` if no matching element is found, which means any attempt to do anything with that result will cause an error.
+
+This means more checks may have to be added to the code.
+
+```javascript
+var example = document.querySelector('.this')
+
+if (example) {
+  // do something
+}
+```
+
+### When to use querySelector and querySelectorAll
+
+In jQuery an element can be found on the page using `.find()`. There are two equivalent functions in JS.
+
+- [querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) finds only the first element that matches the given query.
+- [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) finds all elements that match that query, and returns an array of them.
```

Comment:
I'm going to be picky here - `querySelectorAll()` doesn't return an array but a non-live NodeList. A minor point, as the NodeList can be treated as an array in _most_ instances, but one that I think it worth clarifying as it can trip things up when `forEach()` is used in older browsers.

Reply:
Updated, thanks.